pins/tests/test_versions.py

```diff
@@ -21,6 +21,18 @@ def test_version_from_string():
     assert version.hash == "baf3f"
 
 
+def test_version_from_string_too_many_hyphens():
+    with pytest.raises(
+        PinsVersionError, match="version string can only have 1 '-', but contains 2"
+    ):
+        Version.from_string("20220209T220116Z-baf3f-")
+
+
+def test_version_from_string_too_few_hyphens():
+    with pytest.raises(
+        PinsVersionError, match="version string can only have 1 '-', but contains 0"
+    ):
+        Version.from_string("20220209T220116Zbaf3f")
```

Comment:
```suggestion
        Version.from_string("20220209T220116Zbaf3f")


```

for linting, I messed this up in the merge 😩